docs/source/model_doc/perceiver.rst

```diff
@@ -0,0 +1,228 @@
+the :obj:`d_latents` to :obj:`num_labels`.
+
+This was the idea of the original Perceiver paper. However, it could only output classification logits. In their
+follow-up work, PerceiverIO, they generalized it to let the model also produce outputs of arbitrary size. How, you
+might ask? The idea is actually relatively simple: one defines outputs of an arbitrary size, and then applies
+cross-attention with the last hidden states of the latents, using the outputs as queries, and the latents as keys and
+values.
+
+So let's say you want to perform masked language modeling (BERT-style) with the Perceiver. As the Perceiver's input
+length will not have an impact on the computation time, one can provide raw bytes, providing :obj:`inputs` of length
+2048 to the model. If we now mask out certain of these 2048 tokens, we can define our :obj:`outputs` as being of shape:
+:obj:`(batch_size, 2048, 768)`. Next, we perform cross-attention with the final hidden states of the latents to update
+the :obj:`outputs` tensor. After cross-attention, we still have a tensor of shape :obj:`(batch_size, 2048, 768)`. We
+can then place a regular language modeling head on top, to project the last dimension to the vocabulary size of the
+model, i.e. creating logits of shape :obj:`(batch_size, 2048, 262)` (as Perceiver uses a vocabulary size of 262 byte
+IDs).
+
+
+This model was contributed by `<nielsr> <https://huggingface.co/nielsr>`__. The original code can be found `here
+<https://github.com/deepmind/deepmind-research/tree/master/perceiver>`__.
+
+
+Perceiver specific outputs
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+.. autoclass:: transformers.models.perceiver.modeling_perceiver.PerceiverModelOutput
+    :members:
+
+.. autoclass:: transformers.models.perceiver.modeling_perceiver.PerceiverDecoderOutput
+    :members:
+
+.. autoclass:: transformers.models.perceiver.modeling_perceiver.PerceiverMaskedLMOutput
+    :members:
+
+.. autoclass:: transformers.models.perceiver.modeling_perceiver.PerceiverClassifierOutput
+    :members:
+
+
+PerceiverConfig
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+.. autoclass:: transformers.PerceiverConfig
+    :members:
+
+
+PerceiverTokenizer
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+.. autoclass:: transformers.PerceiverTokenizer
+    :members: build_inputs_with_special_tokens, get_special_tokens_mask,
+        create_token_type_ids_from_sequences, save_vocabulary
+
+
+PerceiverFeatureExtractor
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+.. autoclass:: transformers.PerceiverFeatureExtractor
+    :members: 
+
+
+PerceiverTextPreprocessor
```

Comment:
Agreed. They can be documented like the specific outputs if you really want to, but they shouldn't be in the main init.

Reply:
I believe it can be useful for end-users, because the Perceiver is really mixing and matching lego blocks together. 

For example, one could create a Perceiver for audio classification by defining a new PerceiverModel that uses a PerceiverAudioPreprocessor as input preprocessor and a PerceiverClassificationDecoder as decoder.

The models that are added (PerceiverForImageClassification, PerceiverForOpticalFlow, etc.) just show how you can mix-and-match these pre and postprocessors to make the Perceiver handle various modalities. They aren't as configurable as a ViT or BERT., you cannot easily tweak their attributes without breaking the model. Every Perceiver model is independent and more of an example to showcase what you can do with the Perceiver.

Reply:
Ok, that makes sense. And what about documentation, can I include them in the docs?